src/pages/commits/utils.ts

```diff
@@ -0,0 +1,171 @@
+  tasks: string[];
+  variants: string[];
+  requesters: string[];
+}
+interface MainlineCommitOptions {
+  projectID: string;
+  limit: number;
+  skipOrderNumber: number;
+}
+interface CommitsPageReducerState {
+  filterState: FilterState;
+  mainlineCommitOptions: MainlineCommitOptions;
+}
+
+const getMainlineCommitsQueryVariables = (
+  state: CommitsPageReducerState
+): MainlineCommitsQueryVariables => {
+  const variables = {
+    mainlineCommitsOptions: generateMainlineCommitOptionsFromState(state),
+    buildVariantOptions: generateBuildVariantOptionsFromState(state),
+    buildVariantOptionsForGraph: generateBuildVariantOptionsForGraphFromState(
+      state
+    ),
+    buildVariantOptionsForGroupedTasks: generateBuildVariantOptionsForGroupedTasksFromState(
+      state
+    ),
+    buildVariantOptionsForTaskIcons: generateBuildVariantOptionsForTaskIconsFromState(
+      state
+    ),
+  };
+  return variables;
+};
+
+/** getFilterStatus returns an object containing booleans that describe what filters have been applied. */
+const getFilterStatus = (state: FilterState) => ({
+  hasFilters:
+    state.requesters.length > 0 ||
+    state.statuses.length > 0 ||
+    state.tasks.length > 0 ||
+    state.variants.length > 0,
+  hasRequesterFilter: state.requesters.length > 0,
+  hasStatuses: state.statuses.length > 0,
+  hasTasks: state.tasks.length > 0,
+  hasVariants: state.variants.length > 0,
+});
+
+const generateBuildVariantOptionsFromState = (
+  state: CommitsPageReducerState
+) => {
+  const { filterState } = state;
+  const { statuses, tasks, variants } = filterState;
+
+  const buildVariantOptions = {
+    tasks,
+    variants,
+    statuses,
+  };
+
+  return buildVariantOptions;
+};
+
+const generateBuildVariantOptionsForTaskIconsFromState = (
+  state: CommitsPageReducerState
+): MainlineCommitsQueryVariables["buildVariantOptions"] => {
+  const { filterState } = state;
+  const { hasTasks, hasFilters, hasStatuses } = getFilterStatus(filterState);
+
+  let statusesToShow = FAILED_STATUSES as string[];
+
+  if (hasFilters && !hasTasks && !hasStatuses) {
+    statusesToShow = [];
+  }
+  if (hasStatuses && !hasTasks) {
+    statusesToShow = arrayIntersection(FAILED_STATUSES, filterState.statuses);
+  }
+  if (hasTasks) {
+    statusesToShow = filterState.statuses;
+  }
+
+  const shouldShowIcons = hasTasks || statusesToShow.length > 0;
+  const buildVariantOptions = {
+    tasks: shouldShowIcons ? filterState.tasks : [impossibleMatch],
+    variants: filterState.variants,
+    statuses: shouldShowIcons ? statusesToShow : [],
+  };
+  return buildVariantOptions;
+};
+
+const generateBuildVariantOptionsForGroupedTasksFromState = (
+  state: CommitsPageReducerState
+): MainlineCommitsQueryVariables["buildVariantOptionsForGroupedTasks"] => {
+  const { filterState } = state;
+  const { hasTasks, hasFilters, hasStatuses } = getFilterStatus(filterState);
+  const { statuses } = filterState;
+
+  let shouldShowTasks = false;
```

Comment:
nit. `shouldShowGroupedTasks` might be clearer or `showGroupedTasks` if you want it to be shorter